### PR TITLE
Transport Update

### PR DESCRIPTION
**Pull request by TelegramSam**
Some issues. I will add commentary to the PR.
Signed-off-by: Sam Curren <telegramsam@gmail.com>

### CODE REVIEW

**Comment by TelegramSam** [comment]
Discussion Topic: One-way transport vs Bi-directional, and how to indicate.
Discussion Topic: Websocket plain vs Websocket STOMP

File: transports.md [hunk 1]
```diff
@@ -2,15 +2,20 @@
...
 #### HTTP(S)
 
 HTTP(S) transports are an effective way to send a message to another online agent.
 
 - Messages are transported via HTTP POST.
-- The MIME Type for the POST request is `application/didcomm-enc-env`.
-- The HTTP POST should return a 202 Accepted status code. This indicates that the request was received, but not necessarily processed. Returning a 200 OK status code is allowed.
-- POST requests are considered transmit only by default. No agent messages will be returned in the response. This behavior may be modified with additional signaling.
+- The MIME Type for the POST request is `application/didcomm-enc-env`. (TODO: Declare Mime Types elsewhere, link from here.)
+- The HTTP POST should return a 202 Accepted status code. A successful message receipt MUST return a code in the 2xx HTTP Status Code range.
```
Comment (kdenhartog) [inline transports.md:16]:
```suggestion
- A successful message receipt MUST return a code in the 2xx HTTP Status Code range. It is recommended that a HTTP POST should return a 202 Accepted status code. 
```

**Comment by kdenhartog** [comment]
Minor editorial suggestion, but this looks good. Much more specific and likely to achieve interoperable code I'd think.

File: transports.md [hunk 2]
```diff
@@ -1,19 +1,35 @@
...
-The POST message MIME type is application/didcomm-enc-env.
-A successful response code is 202 (Accepted), but 200 (OK) is also allowed.
+
+HTTP(S) transports are an effective way to send a message to another online agent.
+
+- Messages are transported via HTTP POST.
+- The MIME Type for the POST request is `application/didcomm-enc-env`. (TODO: Declare Mime Types elsewhere, link from here.)
+- The HTTP POST should return a 202 Accepted status code. A successful message receipt MUST return a code in the 2xx HTTP Status Code range.
+- POST requests are transmit only. Messages are only sent from the code that submitted the POST request.
+- HTTP Redirects SHOULD be followed. All redirects are considered temporary until updated within the appropriate DID Document.
```
[the remark this answers is not in the document]
Comment (TelegramSam) [reply transports.md]: change made

**Comment by mccown** [comment]
The temporary redirects feature is a necessary capability to add.  That said, redirects have previously been used as attack vectors (e.g., MiTM) for HTTP.  With that in mind, I wanted to post some links to some corresponding IETF RFCs, so that we can incorporate their considerations into DIDComm.

IETF RFC: # 7231, Section 6.4.7 references the Temporary Redirect (307) status code
https://tools.ietf.org/html/rfc7231#section-6.4.7 

IETF RFC: # 7230, Section 9 provides some security considerations for the message routing
https://tools.ietf.org/html/rfc7230#section-9 

IETF RFC: # 7231, Section 9 provides some security considerations related to the semantics and message content
https://tools.ietf.org/html/rfc7231#section-9

**Comment by dhh1128** [comment]
+1 to merge from me

**Pull request merged**